backstop/tests/color-picker.html

```diff
@@ -0,0 +1,267 @@
+            tabindex="0"
+          ></span>
+
+          <span
+            style="--swatch-color: hsl(95, 71%, 23%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(95, 72%, 30%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(95, 71%, 38%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(95, 71%, 42%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(95, 71%, 46%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(95, 72%, 61%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(95, 71%, 73%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(94, 71%, 84%);"
+            class="iui-color-swatch"
+            tabindex="0"
+            id="test-color-swatch"
+          ></span>
+
+          <span
+            style="--swatch-color: hsl(94, 43%, 93%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(209, 11%, 64%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(0, 100%, 40%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(23, 100%, 50%);"
+            class="iui-color-swatch"
+            tabindex="0"
+          ></span>
+          <span
+            style="--swatch-color: hsl(42, 99%, 60%);"
+            class="iui-color-swatch  iui-active"
+            tabindex="0"
+          ></span>
+        </div>
+      </div>
+    </section>
+
+    <br />
+
+    <h2>Advanced</h2>
+    <section id="demo-advanced">
+      <div class="iui-color-picker iui-advanced">
+        <div class="iui-color-selection-wrapper">
+          <div
+            class="iui-color-field"
+            style="--color: #FFC335;"
+          >
+            <div
+              class="iui-color-dot"
+              style="--left: 79.2157%; --top: 2%; --selected-color: rgb(255, 195, 53);"
+              tabindex="0"
+              id="test-color-dot"
+            ></div>
+          </div>
+          <div class="iui-color-slider">
+            <div
+              class="iui-color-dot iui-white"
+              style="--top: 14%;"
+              tabindex="0"
+            ></div>
+          </div>
+        </div>
+
+
+        <div style="width: 100%;">Saved Colors</div>
```

Comment:
Yea, agree. I think it should have some bottom margin, it looks too close (added g on purpose there :D):
![image](https://user-images.githubusercontent.com/81580355/131318944-5b073d6a-16fb-4b17-973f-63b736761f9b.png)
We might need a new class on this one `iui-saved-color-label` or something to add missing rules (margin, overflow, etc..).

Reply:
Summarize:
- Double check the need of width. If its needed, screenshot would be useful, so we could double check, why this is appearing.
- Top/bottom margin should use baseline spacing. 
- Overflow still missing.

Reply:
I set width because I want the Saved Colors label have its own row, even when there are only a few colors
![image](https://user-images.githubusercontent.com/88331924/131558946-3646bb74-e552-47eb-8a1b-994e0a091607.png)

Without it the colors move up to the same row
![image](https://user-images.githubusercontent.com/88331924/131558912-e522ac42-3155-4ae2-8f9b-40a45cc7e50d.png)

Reply:
Oh I see what you mean now. See this is a situation where my other comment about adding containers would have helped. https://github.com/iTwin/iTwinUI/pull/253#discussion_r697712651

Reply:
Yes, you need new mixins/classes to nest your flex layout. You might be able to get away with just one container, but you might need two if the spacing is weird.
```html
<color-picker> // this will have flex-direction: column
  <selection-container> // this will be a new class with display: flex
    <selection ...>
    <slider ...>
  </selection-container>
  <div>Saved colors</div>
  <color-palette ...>
</color-picker>
```

Or you can leave `width: 100%` if that's too much.

Still need to tackle the other parts: margin values and overflow.

Reply:
I added flex-direction to iui-advanced, saved-colors wrapper mixin, and margins and overflow to iui-saved-colors

Reply:
Is the `saved-colors-wrapper` really needed? Seems it only applies margin-top, which `saved-colors` was applying anyway.

Reply:
I'm not really sure but without it if there are a lot of colors the saved colors label gets covered up for some reason 
![image](https://user-images.githubusercontent.com/88331924/131857886-f1febd14-7089-4fb7-87b4-42b67b45b244.png)

Reply:
If you add `flex-shrink: 0` to `saved-colors`, you can remove the wrapper.

Reply:
I added it and removed saved-colors-wrapper